### ISSUE TITLE
integrate ingnore-file in yaml config

### ISSUE THREAD

**Issue by MiWeiss** [HunterMcGushion/docstr_coverage#40]
This is a follow-up issue to PR #39. 

With #39 we will potentially have two config files. This is a bit much for a single script.  I guess that from a user's perspective, it would be nicer if they could (optionally) put everything in one file.

I'd imagine something where a file with this content... 

> SomeFile method_to_ignore1 method_to_ignore2 method_to_ignore3
> FileWhereWeWantToIgnoreAllSpecialMethods __.+__
> .* method_to_ignore_in_all_files
> a_very_important_view_file ^get$ ^set$ ^post$
> detect_.* get_val.*

...could be replaced in yaml with something along the lines of:

```
custom_ignore_patterns: 
 - "SomeFile method_to_ignore1 method_to_ignore2 method_to_ignore3"
 - "FileWhereWeWantToIgnoreAllSpecialMethods __.+__"
 - ".* method_to_ignore_in_all_files"
 - "a_very_important_view_file ^get$ ^set$ ^post$"
 - "detect_.* get_val.*"
```

The open question would of course be the preference: If both file path and this array are provided in the yaml, which one should be used? Both? My opinion is: **Do nothing. Fail. **, as this is probably unintended. Of course, if the file path is provided as command-line arg; that one should be used, as explained in my review in #39.

What do you think? I'm happy to hear some opinions. 

Also, I may find some time early next week to implement this.

**Comment by HunterMcGushion** [comment]
This is a great idea. Just so others can follow along, this is referring to the `--docstr-ignore-file` option added in https://github.com/HunterMcGushion/docstr_coverage/pull/4. This would be a fantastic addition after we merge #39. 

Regarding what to do if both the filepath and this array are provided in the yaml, I'm not too sure. I think it would also make sense to choose one of the two options as the preferred one, and simply throw up a warning saying that both were given and "option X" is being used. Your suggestion makes a lot of sense too, though. I think either way should be fine as long as we explain that both options are being used, which causes problems in the error/warning message.

**Comment by MiWeiss** [comment]
I'll take care of it once #39 is merged

**Comment by MiWeiss** [comment]
Quick heads-up: This feature is working on my PC, thus a PR can be expected relatively soon.
However, over Christmas, I'll only have a win machine at my disposal; thus, out of pure self-interest, before I open a PR for this issue, I'll do some work on windows compatibility of the test suite ;-)

**Comment by HunterMcGushion** [comment]
Sounds great! Thanks a lot, Michael! Hahaha I would have to disagree with your comment that your selfless work is out of "pure self-interest", but thank you for tackling the issue of Windows compatibility. I rarely use Windows, so I appreciate you taking the lead on this.

**Comment by MiWeiss** [comment]
This is now merged
